kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/parquet/ParquetColumnWriters.java

```diff
@@ -0,0 +1,477 @@
+                return new DecimalLongWriter(colName, fieldIndex, columnVector);
+            }
+            // TODO: Need to support legacy mode where all decimals are written as binary
+            return new DecimalFixedBinaryWriter(colName, fieldIndex, columnVector);
+        } else if (dataType instanceof DateType) {
+            return new DateWriter(colName, fieldIndex, columnVector);
+        } else if (dataType instanceof TimestampType) {
+            return new TimestampWriter(colName, fieldIndex, columnVector);
+        } else if (dataType instanceof ArrayType) {
+            return new ArrayWriter(colName, fieldIndex, columnVector);
+        } else if (dataType instanceof MapType) {
+            return new MapWriter(colName, fieldIndex, columnVector);
+        } else if (dataType instanceof StructType) {
+            return new StructWriter(colName, fieldIndex, columnVector);
+        }
+
+        throw new IllegalArgumentException("Unsupported column vector type: " + dataType);
+    }
+
+    /**
+     * Base class for column writers. Handles the common stuff such as null check, start/stop of
+     * field and delegating the actual writing of non-null values to the subclass.
+     */
+    abstract static class ColumnWriter {
+        protected final String colName;
+        protected final int fieldIndex;
+        protected final ColumnVector columnVector;
+
+        ColumnWriter(String colName, int fieldIndex, ColumnVector columnVector) {
+            this.colName = colName;
+            this.fieldIndex = fieldIndex;
+            this.columnVector = columnVector;
+        }
+
+        void writeRowValue(RecordConsumer recordConsumer, int rowId) {
+            if (!columnVector.isNullAt(rowId)) {
+                recordConsumer.startField(colName, fieldIndex);
+                writeNonNullRowValue(recordConsumer, rowId);
+                recordConsumer.endField(colName, fieldIndex);
+            }
+        }
+
+        /**
+         * Each specific column writer for data type, will implement to call appropriate methods
+         * on the {@link RecordConsumer} to write the non-null value.
+         */
+        abstract void writeNonNullRowValue(RecordConsumer recordConsumer, int rowId);
+    }
+
+    static class BooleanWriter extends ColumnWriter {
+        BooleanWriter(String name, int fieldId, ColumnVector columnVector) {
+            super(name, fieldId, columnVector);
+        }
+
+        @Override
+        void writeNonNullRowValue(RecordConsumer recordConsumer, int rowId) {
+            recordConsumer.addBoolean(columnVector.getBoolean(rowId));
+        }
+    }
+
+    static class ByteWriter extends ColumnWriter {
+        ByteWriter(String name, int fieldId, ColumnVector columnVector) {
+            super(name, fieldId, columnVector);
+        }
+
+        @Override
+        void writeNonNullRowValue(RecordConsumer recordConsumer, int rowId) {
+            recordConsumer.addInteger(columnVector.getByte(rowId));
```

Comment:
It is `getByte` because the vector stores byte values. Parquet has one physical type `int` for `byte`, `short` and `int` logical types. Internally it has a mechanism to encode to save the space on disk.